src/sidebar/util/test/fetch-test.js

```diff
@@ -0,0 +1,83 @@
+import { FetchError, fetchJSON } from '../fetch';
+
+describe('sidebar/util/fetch', () => {
+  describe('fetchJSON', () => {
+    let fakeResponse;
+
+    beforeEach(() => {
+      fakeResponse = {
+        status: 200,
+        json: sinon.stub().resolves({}),
+        get ok() {
+          return this.status >= 200 && this.status <= 299;
+        },
+      };
+      sinon.stub(window, 'fetch').resolves(fakeResponse);
+      window.fetch.resolves(fakeResponse);
+    });
+
+    afterEach(() => {
+      window.fetch.restore();
+    });
+
+    it('fetches the requested URL', async () => {
+      const init = { method: 'GET' };
+      await fetchJSON('https://example.com', init);
+      assert.calledWith(window.fetch, 'https://example.com', init);
+    });
+
+    it('throws a FetchError if `fetch` fails', async () => {
+      window.fetch.rejects(new Error('Fetch failed'));
+
+      let err;
+      try {
+        await fetchJSON('https://example.com');
+      } catch (e) {
+        err = e;
+      }
+
+      assert.instanceOf(err, FetchError);
+      assert.include(err.message, 'Network request failed: Fetch failed');
```

Comment:
It would be important to know the which `URL` failed.

Reply:
I have added the request URL to `FetchError`'s properties.